memphis/consumer.py

```diff
@@ -156,6 +159,21 @@ async def main(host, username, password, station):
         
         """
         messages = []
+
+        if prefetch:
+            if len(self.cached_messages) >= batch_size:
+                messages = self.cached_messages[:batch_size]
+                self.cached_messages = self.cached_messages[batch_size:]
+                number_of_messages_to_prefetch = batch_size * 2 - batch_size  # calculated for clarity
+            elif self.cached_messages:
+                number_of_messages_to_prefetch = batch_size * 2 - len(self.cached_messages)
+                messages = self.cached_messages
+                self.cached_messages = []
+            else:
+                number_of_messages_to_prefetch = batch_size * 2
+            self.load_messages_to_cache(number_of_messages_to_prefetch)
+            return messages
```

Comment:
If prefetch = true and self.cached_messages is empty then  messages is empty and being returned - instead should fetch messages to return and then load messages to cache